dispatch/apps/events/models.py

```diff
@@ -0,0 +1,61 @@
+import urllib, os
+
+from django.core.files import File
+from django.core.validators import RegexValidator
+
+from django.db.models import (
+    Model, DateTimeField, CharField, TextField, PositiveIntegerField,
+    ImageField, FileField, BooleanField, ForeignKey, ManyToManyField,
+    SlugField, EmailField, SET_NULL)
+
+class Event(Model):
+    title = CharField(max_length=255)
+    description = TextField(max_length=1000)
+    host = CharField(max_length=255)
+
+    image = ImageField(upload_to='events/', null=True, blank=True)
+
+    start_time = DateTimeField(null=True, blank=True)
+    end_time = DateTimeField(null=True, blank=True)
+    location = CharField(max_length=500)
+    address = CharField(max_length=500, null=True, blank=True)
+
+    CATEGORY_CHOICES = (
+        ('sports', 'Sports'),
+        ('music', 'Music'),
+        ('academic', 'Academic'),
+        ('party', 'Party'),
+        ('business', 'Business'),
+        ('ceremony', 'Ceremony'),
+        ('workshop', 'Workshop'),
+        ('other', 'Other')
+    )
+
+    category = CharField(max_length=20, choices=CATEGORY_CHOICES)
+
+    facebook_url = TextField(max_length=500, null=True, blank=True)
+
+    facebook_image_url = TextField(max_length=500, null=True, blank=True)
+
+    is_submission = BooleanField(default=False, blank=True)
+
+    submitter_email = EmailField(null=True)
+
+    phone_regex = RegexValidator(regex=r'(\+?1-)?\(?([0-9]){3}(\)?)(|\s?|-?|\.?)([0-9]){3}(\)?|\s?|-?|\.?)([0-9]{4})$', message='Please enter a valid phone number')
+
+    submitter_phone = CharField(max_length=15, validators=[phone_regex], null=True)
+
+    is_published = BooleanField(default=False)
+
+    def cacheimage(self):
+        """Store image locally if we have a facebook url"""
+
+        if self.facebook_image_url and not self.image:
+
+            result = urllib.urlretrieve(self.facebook_image_url)
```

Comment:
After reading through their documentation and playing around with the code I couldn't get it to work. Maybe I was implementing it incorrectly, but I was getting some weird responses. I think `urllib` will be easier